### PR TITLE
[MIG] l10n_de_country_states: Migration to 10.0

### PR DESCRIPTION
**Pull request by chienandalu**
Depends on:

- [x] https://github.com/OCA/partner-contact/pull/461 base_country_state_translatable

cc @Tecnativa

### CODE REVIEW

File: l10n_de_country_states/README.rst
Comment (pedrobaeza) [inline l10n_de_country_states/README.rst]: Include license flag as in the README template: https://raw.githubusercontent.com/OCA/maintainer-tools/master/template/module/README.rst

File: l10n_de_country_states/README.rst
Comment (pedrobaeza) [inline l10n_de_country_states/README.rst]: Use short version according template.

File: l10n_de_country_states/__manifest__.py
Comment (pedrobaeza) [inline l10n_de_country_states/__manifest__.py]: Put Tecnativa instead on Antiun

File: l10n_de_country_states/models/res_country.py
Comment (pedrobaeza) [inline l10n_de_country_states/models/res_country.py]: This module shouldn't do this change. Why it's needed?

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: l10n_de_country_states/models/res_country.py
Comment (OSevangelist) [reply l10n_de_country_states/models/res_country.py]: HI Pedro,
i will check back with Nikki who did that contribution in the first place and get all open issues done this week. Thank you for your patience
Cheers Frederik

Comment (pedrobaeza) [reply l10n_de_country_states/models/res_country.py]: @OSevangelist any news?

Comment (OSevangelist) [reply l10n_de_country_states/models/res_country.py]: Sorry, forgot to do that. Will check back with Nikki now.

Comment (rami-wafaie) [reply l10n_de_country_states/models/res_country.py]: The Module was originall developed for opernerp-7.0 and migrated to Odoo 8 it adds the German states to the res.country.state so it can be selected when editing the address of  a contact, employee, user....
I tested the module on Odoo 9 and did the required changes and made a PR:
https://github.com/OCA/l10n-germany/pull/20
but till now it has not been reviewed.
For Odoo 10 I have not tested it yet

Comment (pedrobaeza) [reply l10n_de_country_states/models/res_country.py]: So can we remove this change on v10 migration?

Comment (rami-wafaie) [reply l10n_de_country_states/models/res_country.py]: No we need the translation, the German states names are different in English and German, that's why we need the 'name' field translatable

Comment (pedrobaeza) [reply l10n_de_country_states/models/res_country.py]: Yeah, but that's something not to be done here in this module, but in a generic one. It's also interesting that you propose in Odoo master the change of the field to translatable.

Comment (OSevangelist) [reply l10n_de_country_states/models/res_country.py]: I would tentatively agree on what you've mentioned @pedrobaeza but do we have any *rather strict* convention how we "architect" modules in the OCA or any "written rule" of do's and don'ts

Comment (pedrobaeza) [reply l10n_de_country_states/models/res_country.py]: Not explicitly, but as it's something generic (you are changing translatable property for all the country states, not only for Germany), I would say to make it generic via a module `base_country_state_translatable` or similar and depend this module on that one.

Comment (OSevangelist) [reply l10n_de_country_states/models/res_country.py]: @pedrobaeza Agree. Maybe we find some time in October to discuss the "optimal" relation between generalizability and specificity, since that is at least to my point of view something that is comparably poorly handled within the Odoo ecosystem (and many other OS frameworks). Sorry for kind of offtopic

Comment (rami-wafaie) [reply l10n_de_country_states/models/res_country.py]: @pedrobaeza OK I will take care of it

**Comment by rami-wafaie** [comment]
@chienandalu can you please remove the changes in res.country from this module, I will create a separate module that make the name field in res.country translatable and add the german translation there

**Comment by chienandalu** [comment]
@rami-wafaie Ok. I'd need the future module's name so I can set the dependency ant leave this ready to merge once the other part is ready.

**Comment by rami-wafaie** [comment]
@chienandalu Thanks I will notify you when it is ready

**Comment by rami-wafaie** [comment]
The second module is created: ' base_country_state_translatable'
Here is the PR: https://github.com/OCA/l10n-germany/pull/25
https://github.com/OCA/l10n-germany/pull/25

**Comment by rami-wafaie** [comment]
New Modul is created: 'base_country_state_translatable'
Here is the link to the PR: https://github.com/OCA/partner-contact/pull/461

**Comment by chienandalu** [comment]
Thanks @rami-wafaie :) Changes done to adapt this module to the new one

**Comment by pedrobaeza** [comment]
@chienandalu please add oca_dependencies.txt file with partner-contact for correct CIs.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
Include Installation section telling about partner-contact

**Comment by chienandalu** [comment]
@pedrobaeza All green

File: l10n_de_country_states/README.rst
Comment (pedrobaeza) [inline l10n_de_country_states/README.rst]: s/patner-contact/partner-contact

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
@rami-wafaie @OSevangelist any objection in merging this?

**Comment by OSevangelist** [comment]
No seem fine. Go from my side

**Comment by chienandalu** [comment]
@pedrobaeza I think is all done here

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**
